---
# tasks file for pvwa registration

- name: Registration Block
  block:

    - set_fact:
        pvwa_registration_configure_instance: true
        pvwa_registration_configure_vault: true
        pvwa_registration_register_instance: true
      when: pvwa_registration

    - name: Assign PVWA IP
      set_fact:
        pvwa_url: "https://{{ ansible_host }}/PasswordVault"
      when: pvwa_url is not defined

    - name: Stop report service
      win_service:
        name: "CyberArk Scheduled Tasks"
        state: stopped
      when:
        - pvwa_registration_configure_instance

    - name: Run {{ action }}
      include_tasks: pvwa_registration_actions.yml
      with_items:
        - { action: ConfigureInstance, condition: pvwa_registration_configure_instance }
        - { action: ConfigureVault, condition: pvwa_registration_configure_vault }
        - { action: RegisterInstance, condition: pvwa_registration_register_instance }
      vars:
        - action: "{{ item.action }}"
        - condition: "{{ item.condition }}"

    - name: Check that you can connect (GET) to pvwa and it returns a status 200
      win_uri:
        url: "https://127.0.0.1/PasswordVault/api/server"
        validate_certs: no
        timeout: 60
      register: pvwa_server_info
      until: pvwa_server_info.status_code == 200
      retries: 12
      delay: 10
      when:
        - pvwa_registration_register_instance

    - name: Start report service
      win_service:
        name: "CyberArk Scheduled Tasks"
        state: started
      when:
        - pvwa_registration_register_instance

    - set_fact:
        pvwa_registered: true
      when:
        - pvwa_registration_register_instance

  rescue:

    - fail:
        msg: 'ERROR: Registration failed. For more info check {{ lookup("config", "DEFAULT_LOG_PATH") | dirname }}/pvwa/{{ inventory_hostname }}_registration.log'
